File: tools/improve/tests/original/maudec/py/sorts.py
import enum

class Bar(enum.Enum):
	F = 0  # 
	G = 1  # 


def hasSort_Foo(obj: Bar) -> bool:
	match obj:
		case (Bar.F, ):
			return True

		case _:
			return False



def isFoo1(B: Bar) -> bool:
	return hasSort_Foo(B)


def isFoo2(B: Bar) -> bool:
	match B:
		case F if hasSort_Foo(F):
			return True

		case _:
			return False



def isPositive(Z: int) -> bool:
	match Z:
		case N if N > 0:
			return True

		case _:
			return False

File: tools/improve/tests/original/maudec/rust/sorts.rs


#[derive(Clone, PartialEq, PartialOrd, Debug)]
enum Bar {
	F,
	G,
}

fn hasSort_Foo(obj: &Bar) -> bool {
	match obj {
		Bar::F => true,
		_ => false,
	}
}

fn isFoo1(b: &Bar) -> bool {
	hasSort_Foo(b)
}

fn isFoo2(b: &Bar) -> bool {
	match b {
		f if hasSort_Foo(f) => true,
		_ => false,
	}
}

fn isPositive(z: i32) -> bool {
	match z {
		n if n > 0 => true,
		_ => false,
	}
}

